lib/manager/gitlabci/extract.js

```diff
@@ -0,0 +1,91 @@
+const { splitImageParts } = require('../docker/extract');
+
+module.exports = {
+  extractDependencies,
+};
+
+function extractDependencies(content) {
+  const deps = [];
+  try {
+    const lines = content.split('\n');
+    for (let lineNumber = 0; lineNumber < lines.length; lineNumber += 1) {
+      const line = lines[lineNumber];
+      const imageMatch = line.match(/^\s*image:\s*'?"?([^\s]+)'?"?\s*$/);
+      if (imageMatch) {
+        logger.trace(`Matched image on line ${lineNumber}`);
+        const currentFrom = imageMatch[1];
+        const {
+          dockerRegistry,
+          depName,
+          currentTag,
+          currentDigest,
+          currentDepTagDigest,
+          currentDepTag,
+        } = splitImageParts(currentFrom);
+        logger.info(
+          { dockerRegistry, depName, currentTag, currentDigest },
+          'Docker Compose image'
+        );
+        deps.push({
+          depType: 'image',
+          lineNumber,
+          currentFrom,
+          currentDepTagDigest,
+          dockerRegistry,
+          currentDepTag,
+          currentDigest,
+          depName,
+          currentTag,
+        });
```

Comment:
At least as of my testing with last night's code, the `deps` object does not contain a `currentValue` property as required by the `package.js` file in the Docker manager.

Since Docker has been refactored several hours ago I guess I'll need to re-test.

Reply:
@hbetts sorry, this PR is in need of some refactoring due to changes to the mainline. Today’s refactor of Docker lookups will make this one even more out of date, but I should be able to do the fixes soon if there’s now someone to test it again :)

Reply:
So I definitely want this feature for my GitLab projects. I want to know I can run a CI job multiple times and always get the same exact Docker image.

Sorry I hadn't had much time to look into this earlier. If you refactor it, I will definitely test it (There just might be a little delay as it sits behind slightly higher priority items)